great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -802,7 +802,7 @@ class TableAsset(_SQLAsset):
 
     # Instance fields
     type: Literal["table"] = "table"
-    table_name: str = pydantic.Field(
+    table_name: Union[str, sqlalchemy.quoted_name] = pydantic.Field(
```

Comment:
I wonder if this can always be `sqlalchemy.quoted_name`?

Reply:
@billdirks I am not sure -- it seems to me that the user first provides a string, and then we -- in the validator -- check it, and if it is not `None`, then we quote it.  So the question is: how many times does the field get set?  Twice or once?  Thanks!

Reply:
@billdirks The verdict is in!  While `mypy` accepts the main sql_datasource.py module either way, the test module shows a type violation in PyCharm, because we submit a string valued table name to the `TableAsset()` constructor.  So I am leaving the `Union` in.  Thanks!